### PR TITLE
#164139663 Implement functionality to report articles

### PR DESCRIPTION
**Pull request by codinger41**
#### What does this PR do?
Consumes the endpoint to report articles

#### Description of Task to be completed?
I created an action to report articles, which gets fired by users when they use the `submit` button on the report modal after entering the reason they're reporting the article
#### How should this be manually tested?
- Visit the single article page, find the options icon below the article body, click the button, it'll open a dropdown menu, click the menu to open the report modal.
- On that modal, select one of the options for reasons and enter additional information then click the submit button.
#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/164139663
#### Questions:
N/A

### CODE REVIEW

**Review by kleva-j — DISMISSED** [review]

**Review by olajide1234 — DISMISSED** [review]

**Review by kleva-j — DISMISSED** [review]

File: src/components/reportArticle/Index.jsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
+/* eslint-disable react/jsx-no-bind */
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Button, Modal, Form, Row, Col, Alert, Spinner } from 'react-bootstrap';
+import { reportArticle } from '../../redux/actions/reports';
+
+const timeout = ms => new Promise(res => setTimeout(res, ms));
```
Comment (9jaswag) [inline src/components/reportArticle/Index.jsx]: @leksyib why are we setting time out?

File: src/views/Article/Index.jsx [hunk 2]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable jsx-a11y/click-events-have-key-events */
```
Comment (9jaswag) [inline src/views/Article/Index.jsx]: @leksyib why are we disabling so much eslint rules?

**Review by 9jaswag — CHANGES_REQUESTED** [review]

File: src/components/reportArticle/Index.jsx (hunk 1 above)
Comment (codinger41) [reply src/components/reportArticle/Index.jsx]: @9jaswag  We want to close the modal after some seconds if the reporting is successful so the user can see the success message. We don't want to close it immediately.

File: src/views/Article/Index.jsx (hunk 2 above)
Comment (codinger41) [reply src/views/Article/Index.jsx]: @9jaswag This rule is looking for a key event prop that we're not ready to pass, and we do not want to pass an empty function.

File: src/components/reportArticle/Index.jsx (hunk 1 above)
Comment (9jaswag) [reply src/components/reportArticle/Index.jsx]: remember to fix this

File: src/redux/actions/reports.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/* eslint-disable import/prefer-default-export */
+import axios from 'axios';
+
+const apiurl = 'https://vidar-ah-backend-production.herokuapp.com/api/v1';
+
+export const reportArticle = (reportData, articleSlug) => async () => {
+  try {
+    const token = localStorage.getItem('token');
+    const { data } = await axios.post(
+      `${apiurl}/report/${articleSlug}`,
```
Comment (9jaswag) [inline src/redux/actions/reports.js]: @leksyib since this is a post, we can add the slug to the request body

**Review by 9jaswag — CHANGES_REQUESTED** [review]

**Review by olajide1234 — APPROVED** [review]

**Review by kleva-j — APPROVED** [review]

**Review by 9jaswag — APPROVED** [review]

**Pull request merged**